sdk/identity/src/token_credentials/default_credentials.rs

```diff
@@ -158,3 +161,7 @@ impl azure_core::TokenCredential for DefaultCredential {
             .map_err(|error| azure_core::Error::GetTokenError(Box::new(error)))
     }
 }
+
+fn format_aggregate_error(errors: &Vec<DefaultCredentialError>) -> String {
+    errors.iter().map(|error| format!("\n{}", error)).collect()
```

Comment:
```suggestion
    errors.iter().map(String::to_string).join("\n")
```

Reply:
This line doesn't seem to work exactly like your suggestion. I have it working with this: 
`
errors
        .iter()
        .map(|error| format!("{}", error))
        .collect::<Vec<String>>()
        .join("\n")
`
Is there a better way to write this?

Reply:
AH yes you're right.  I think this should work.

```rust
errors.iter().map(String::to_string).collect::<String>
```